docs/architecture/adr-008-app-caller-cbs/adr-008-app-caller-cbs.md

```diff
@@ -261,32 +277,42 @@ func (im IBCModule) OnAcknowledgementPacket(
     unmarshal(acknowledgement, ack)
 
     // send acknowledgement to original actor
-    acc := k.getAccount(ctx, cbPacketData.GetSrcCallbackAddress())
+    acc := k.getAccount(ctx, cbPacketData.GetSourceCallbackAddress())
     ibcActor, ok := acc.(IBCActor)
     if ok {
         gasLimit := getGasLimit(ctx, cbPacketData)
 
-        // create cached context with gas limit
-        cacheCtx, writeFn := ctx.CacheContext()
-        cacheCtx = cacheCtx.WithGasLimit(gasLimit)
+
+        handleCallback := func() error {
+            // create cached context with gas limit
+            cacheCtx, writeFn := ctx.CacheContext()
+            cacheCtx = cacheCtx.WithGasLimit(gasLimit)
         
-        defer func() {
-            if e := recover(); e != nil {
-                log("ran out of gas in callback. reverting callback state")
-            } else {
-                // only write callback state if we did not panic during execution
-                writeFn()
+            defer func() {
+                if e := recover(); e != nil {
+                    log("ran out of gas in callback. reverting callback state")
+                } else if err == nil {
+                    // only write callback state if we did not panic during execution
+                    // and the error returned is nil
+                    writeFn()
+                }
```

Comment:
The previous pseudo code was incorrect as it didn't handle where the error returned was nil or not. There's also a more subtle issue that you only want to catch the panic of the function call using the `cacheCtx`, a panic performed using the regular context should not be caught as we need to panic to inform baseapp that the transaction execution ran out of gas. For this reason I generate an anonymous function to perform the callback using the `cacheCtx`